### PR TITLE
Feature iid issue #15

### PR DESCRIPTION
**Pull request by TobiasRasbold**
Hi @mansenfranzen  
Issue #15 should be done with this PR.

- IntervalIdentifier is able to use only 'marker_start' for creating intervals
- 'transform'- functions in VectorizedCumSum is enhanced for pyspark and pandas
- adding tests 'test_identical_start_end_marker' for pyspark and pandas

### CODE REVIEW

File: src/pywrangler/pandas/wranglers/interval_identifier.py [hunk 1]
```diff
@@ -187,22 +187,30 @@ def _transform(self, series: pd.Series) -> List[int]:
 
         """
 
-        # get boolean series with start and end markers
-        bool_start = series.eq(self.marker_start)
-        bool_end = series.eq(self.marker_end)
+        if self._naive_algorithm:
+            # when series equals marker_column return True
+            bool_start = series.eq(self.marker_start)
```
Comment (mansenfranzen) [inline src/pywrangler/pandas/wranglers/interval_identifier.py]: Same line of code appears in line 199 - DRY. There should be no need to indent the entire code block with if/else. Instead, the `if self._naive_algorithm:` could be used right after line 199 and return the result `return bool_start.cumsum()` right away. No need for `else` when `return` already exits the function.

File: src/pywrangler/pyspark/wranglers/interval_identifier.py [hunk 2]
```diff
@@ -71,31 +71,45 @@ def transform(self, df: DataFrame) -> DataFrame:
...
-        renumerate_adjusted = F.when(bool_valid, renumerate).otherwise(0)
-
-        # ser_id needs be created temporarily for renumerate_adjusted
-        return df.withColumn(self.target_column_name, ser_id) \
-            .withColumn(self.target_column_name, renumerate_adjusted)
+        if self._naive_algorithm:
+            # identical marker for start and end
+            # get boolean series with start and end markers
+            marker_col = F.col(self.marker_column)
+            bool_start = (marker_col == self.marker_start).cast("integer")
```
Comment (mansenfranzen) [inline src/pywrangler/pyspark/wranglers/interval_identifier.py]: Line 77/78 are identical with line 89/90 - DRY. Similar to pandas `VectorizedCumSum`, the `if` clause could be placed right after line 90. It could be also meaningful to separate the naive (identical start/end markers) and more complex (different start/end markers) into different methods to increase readability.

File: src/pywrangler/wranglers.py [hunk 3]
```diff
@@ -2,12 +2,14 @@
...
 
 """
-
+from typing import Any
 
 from pywrangler.base import BaseWrangler
 from pywrangler.util import sanitizer
 from pywrangler.util.types import TYPE_ASCENDING, TYPE_COLUMNS
 
+NULL = object()
```
Comment (mansenfranzen) [inline src/pywrangler/wranglers.py]: Thinking about the NULL value, a true NULL value (e.g. in spark) could be a valid marker value, too. Therefore, the name should not be NULL but rather `VALUENOTSET` which should be more concise. Nevertheless, values like `None` are `NULL` are edge case. Anyway, I believe it's worth allowing those values because it is unclear how future computation engines will treat any of these values.

File: tests/pyspark/conftest.py [hunk 4]
```diff
@@ -1,6 +1,7 @@
 """pytest configuration
 
 """
+import multiprocessing
```
Comment (mansenfranzen) [inline tests/pyspark/conftest.py:4]: After a rebase, this should be already present.

File: tests/pandas/wranglers/test_interval_identifier.py [hunk 5]
```diff
@@ -198,3 +198,18 @@ def test_no_groupby_order_columns(test_case, wrangler, groupby_order):
 
     test_output = wrangler_instance.fit_transform(test_input)
     assert_frame_equal(test_output, expected)
+
+
+@pytest.mark.parametrize(**MARKERS_KWARGS)
+def test_identical_start_end_marker(marker):
```
Comment (mansenfranzen) [inline tests/pandas/wranglers/test_interval_identifier.py]: Implementation for the `NaiveIterator` should be added, too.

File: tests/pyspark/wranglers/test_interval_identifier.py [hunk 6]
```diff
@@ -233,3 +233,25 @@ def test_no_groupby_order_columns(test_case, wrangler, groupby_order,
     else:
         test_output = wrangler_instance.fit_transform(test_input)
         assert_pyspark_pandas_equality(test_output, expected)
+
+
```
Comment (mansenfranzen) [inline tests/pyspark/wranglers/test_interval_identifier.py]: This test should be run against all available wrangler implementations.
`@pytest.mark.parametrize(**WRANGLER_KWARGS)`

File: tests/pandas/wranglers/test_interval_identifier.py (hunk 5 above)
Comment (mansenfranzen) [reply tests/pandas/wranglers/test_interval_identifier.py]: `@pytest.mark.parametrize(**WRANGLER_KWARGS)` should be added.

File: tests/test_data/interval_identifier.py [hunk 7]
```diff
@@ -354,3 +354,24 @@ def multiple_groupby_order_columns_with_invalids(start, end, noise,
             [4,      2,      2,        2,        end,    0]]
 
     return _return_dfs(data, target_column_name, shuffle=shuffle)
+
```
Comment (mansenfranzen) [inline tests/test_data/interval_identifier.py:357]: One more test with non-sorted order would be nice.

**Review by mansenfranzen — CHANGES_REQUESTED** [review]
Great PR :-)

**Review by mansenfranzen — APPROVED** [review]
